bootstrap/internal/controller/agentbootstrapconfig_controller.go

```diff
@@ -122,6 +121,12 @@ func (r *AgentBootstrapConfigReconciler) Reconcile(ctx context.Context, req ctrl
 		}
 		return ctrl.Result{}, err
 	}
+
+	// Ignore any agent bootstrapconfig being deleted
+	if !config.DeletionTimestamp.IsZero() {
+		return ctrl.Result{}, nil
```

Comment:
Theoretically, the metal3machine takes care of deprovisioning the BMH. Not sure what we should do about the agent? 
If it's a control plane node, we probably wouldn't allow detaching until the AgentControlPlane gets deleted...